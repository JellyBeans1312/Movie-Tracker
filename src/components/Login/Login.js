import React from 'react';
import './Login.css'
import PropTypes from 'prop-types';

export const Login = ({ email, password, handleChange, handleSubmit}) => {
  return (
    <section>
      <form className='log-in-form' onSubmit={handleSubmit}>
        <input className='log-in-email'
          type="text"
          value={email}
          name="email"
          placeholder='example@example.com'
          onChange={handleChange}
          />
        <input className='log-in-password'
          type="password"
          value={password}
          name="password"
          placeholder='Password'
          onChange={handleChange}
          />
        <button className='log-in-btn'>Login</button>
      </form>
    </section>
     )
  };

  Login.propTypes = {
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired
  }