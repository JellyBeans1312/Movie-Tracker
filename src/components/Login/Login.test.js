import { shallow } from 'enzyme';
import React from 'react';
import { Login } from './Login';

describe('Login', () => {
  let wrapper;
  let instance;
  let handleChange;
  let handleSubmit;

  beforeEach(() => {
    handleChange = jest.fn();
    handleSubmit = jest.fn();
    wrapper = shallow(<Login handleChange={handleChange} handleSubmit={handleSubmit}/>);
    instance = wrapper.instance();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should invoke handleSubmit on submit', () => {
    const e = {
      preventDefault: jest.fn()
    }
    wrapper.find('form').simulate('submit');
    
    expect(wrapper.instance().handleSubmit(e));
  });

  it.skip('should invoke handleChange on change', () => {
    wrapper.find(".log-in-email").simulate('change', {value: 'whatever'})
    console.log(wrapper.length);
    expect(instance.handleChange());
  });
});