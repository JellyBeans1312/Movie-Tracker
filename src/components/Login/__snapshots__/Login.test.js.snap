// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<section>
  <form
    className="log-in-form"
    onSubmit={[MockFunction]}
  >
    <input
      className="log-in-email"
      name="email"
      onChange={[MockFunction]}
      placeholder="example@example.com"
      type="text"
    />
    <input
      className="log-in-password"
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      type="password"
    />
    <button
      className="log-in-btn"
    >
      Login
    </button>
  </form>
</section>
`;
